### DISCUSSION TITLE
STM32H7B3I_DK board: HOWTO build micropython

### DISCUSSION THREAD

**Discussion by luccons** [micropython/micropython#14139] [STM32 / Pyboard]
Hi all,

I've followed your guide in order to build and install micropython binary firmware for STM32H7B3I_DK board but it seems I miss some steps.
Could you recap all the necessary steps to complete the build for this board?

Many thanks and best regards,

Francesco

**Comment by davefes** [comment]
Do you get errors during the build process, errors uploading the firmware or ??

**Comment by luccons** [comment]
Hi Dave,

I don't get errors during build process and I suppose that upload process about firmware has been successfully completed, I say that because I've installed your binary firmware (released on micropython website) and it works.

**Reply by davefes** [reply]
I have recently been using the build process here:
https://github.com/nspsck/STM32F411CEU6_BlackPill_Micropython/tree/main
maybe that helps.

Is your firmware .hex file similar in size to the Micropython .hex file?

**Comment by luccons** [comment]
I've followed your guide but my firmware.hex size is lower than Micropython one. What should I enable further during build process?

**Reply by davefes** [reply]
I find they are slightly different, guess it depends on the machine you are building on.  Post a link to the specific guide you are using.  

Maybe someone who has experience on this board would be helpful.

**Comment by luccons** [comment]
At first I've installed in my host pc (featured with Ubuntu 22.04 LTS) the arm toolchain:
 `sudo apt-get -y install gcc-arm-none-eabi build-essential`

Later I've downloaded micropython source code and switched to branch v1.22.2:
`git clone https://github.com/micropython/micropython.git`
`cd micrpython`
`git checkout 'v1.22.2'`
`git submodule update --init --recursive`

Build mpy-cross:
`cd mpy-cross`
`make -j`
Build firmware for board STM32H7B3I_DK and write it on internal flash storage:
`cd ../ports/stm32`
`make -j BOARD=STM32H7B3I_DK`
`st-flash erase`
`st-flash --format ihex write build-STM32H7B3I_DK/firmware.hex`

**Reply by davefes** [reply]
@luccons  The only other things I can suggest:

Do a `make clean BOARD=STM32H7B31_DK` after your `cd ../ports/stm32`

I use STM32Cube Programmer to do the erase and upload.

Sorry, don't think I can help further.

**Reply by IHOXOHI** [reply]
Hi,
Do you have tried to use the firmware from micropython download page, with just the cube programmer software?
I have mine which works, but without the display...
I tried to find a lib for that, without success...

**Reply by luccons** [reply]
Hi,
below you can find a valid micropython build procedure for board STM32H7B31_DK:
```
cd ports/mboot
make -C mboot BOARD=STM32H7B3I_DK
cd ..
BUILDING_MBOOT=1 make BOARD=STM32H7B3I_DK
st-flash erase
st-flash --format ihex write build-STM32H7B3I_DK/firmware.hex
```
